### PR TITLE
Handle signals in 'ssc(1)' to shutdown child app process

### PR DESCRIPTION
**Pull request by jwerle**
We were no longer handling signals with the move to `SSC::Process` from `SSC::exec`. This pull request introduces a signal handler.

### CODE REVIEW

File: src/cli/cli.cc [hunk 1]
```diff
@@ -142,6 +142,23 @@ inline String prefixFile () {
...
+static Process* appProcess = nullptr;
+
+void signalHandler (int signal) {
+  if (appProcess != nullptr) {
+    auto pid = appProcess->getPID();
+    appProcess->kill(pid);
+    delete appProcess;
+    appProcess = nullptr;
+  } else if (appPid > 0) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(32));
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:155]: nit: indentation

**Review by chicoxyzzy — APPROVED** [review]

File: src/cli/cli.cc (hunk 1 above)
Comment (chicoxyzzy) [reply src/cli/cli.cc:155]: nvm it was a GitHub glitch

File: src/cli/cli.cc [hunk 2]
```diff
@@ -142,9 +142,27 @@ inline String prefixFile () {
...
+static std::atomic<int> appStatus = 0;
+static std::mutex appMutex;
+
+void signalHandler (int signal) {
+  appStatus = signal;
+
+  if (appProcess != nullptr) {
+    auto pid = appProcess->getPID();
+    appProcess->kill(pid);
+  } else if (appPid > 0) {
```
Comment (mribbons) [inline src/cli/cli.cc:156]: WIN32 doesn't have `kill` in `signal.h`, we need the following:
```cpp
socket/src/cli/cli.cc:157:5: error: use of undeclared identifier 'kill'
    kill(appPid, signal);
    ^
```

```cpp
#ifndef _WIN32
  else if (appPid > 0) {
    kill(appPid, signal);
    appPid = 0;
  } 
#endif
```

**Review by mribbons — CHANGES_REQUESTED** [review]

**Comment by mribbons** [comment]
Aside from the change I requested, this PR works great @jwerle!

**Review by mribbons — APPROVED** [review]
Excellent :tada:

**Pull request merged**